include/boost/geometry/algorithms/detail/envelope/linestring.hpp

```diff
@@ -144,19 +154,27 @@ struct envelope_linear_on_spheroid
                 typename coordinate_system<Box>::type::units
             >::apply(longitude_intervals, lon_min, lon_max);
 
-        assign_values(mbr, lon_min, lat_min, lon_max, lat_max);
+        geometry::set<min_corner, 0>(mbr, lon_min);
+        geometry::set<min_corner, 1>(mbr, lat_min);
+        geometry::set<max_corner, 0>(mbr, lon_max);
+        geometry::set<max_corner, 1>(mbr, lat_max);
+
+        dispatch::envelope
+            <
+                Linear, 2, DimensionCount
+            >::apply(linear, mbr);
```

Comment:
The problem may be indeed the Segments going though the pole so having `longitude[i+1] == longitude[i] + 180` because then we don't know if the Box should be expanded left or right. Btw, how are such Segments handled now? If the intervals of Segments are gathered first then I'm guessing that there is only one possible representation. But this representation of Segment's Envelope may be not correct for the Linestring since the complementary Box could be better, on the opposite side of the globe.